x/axelarnet/module.go

```diff
@@ -331,53 +331,76 @@ func (am AppModule) OnTimeoutPacket(
 		return err
 	}
 
-	return setTransferFailed(ctx, am.keeper, port, channel, sequence)
+	return setRoutedPacketFailed(ctx, am.keeper, am.nexus, port, channel, sequence)
 }
 
 // GetAppVersion implements the ChannelKeeper interface
 func (am AppModule) GetAppVersion(ctx sdk.Context, portID string, channelID string) (string, bool) {
 	return am.channel.GetAppVersion(ctx, portID, channelID)
 }
 
-// returns true if mapping exits
+// returns the transfer id and delete the existing mapping
 func getSeqIDMapping(ctx sdk.Context, k keeper.Keeper, portID, channelID string, seq uint64) (nexus.TransferID, bool) {
 	defer k.DeleteSeqIDMapping(ctx, portID, channelID, seq)
 
 	return k.GetSeqIDMapping(ctx, portID, channelID, seq)
 }
 
-func setTransferFailed(ctx sdk.Context, k keeper.Keeper, portID, channelID string, seq uint64) error {
+// returns the general message id and delete the existing mapping
+func getSeqMessageIDMapping(ctx sdk.Context, k keeper.Keeper, portID, channelID string, seq uint64) (nexus.MessageID, bool) {
+	defer k.DeleteSeqMessageIDMapping(ctx, portID, channelID, seq)
+
+	return k.GetSeqMessageIDMapping(ctx, portID, channelID, seq)
+}
+
+func setRoutedPacketCompleted(ctx sdk.Context, k keeper.Keeper, n types.Nexus, portID, channelID string, seq uint64) error {
+	// check if the packet is Axelar routed cross chain transfer
 	transferID, ok := getSeqIDMapping(ctx, k, portID, channelID, seq)
-	if !ok {
-		return nil
+	if ok {
+		events.Emit(ctx,
+			&types.IBCTransferFailed{
```

Comment:
emit Failed in setCompleted function? why

Reply:
good catch..